### PR TITLE
chore: make EscrowPacketFee private

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Modified `EscrowPacketFee` to be unexported - `escrowPacketFee`
- Moved `IsFeeEnabled` checks to `msg_server.go`
- Removed all references to `EscrowPacketFee` in tests. Explicitly set fees in escrow and send coins to module account
- Updated `PayPacketFee` and `PayPacketFeeAsync` tests to accommodate unexported `escrowPacketFee`

closes: #1059 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/29-fee/keeper/keeper_test.go [hunk 1]
```diff
@@ -98,11 +98,14 @@ func (suite *KeeperTestSuite) TestFeesInEscrow() {
 	packetID := channeltypes.NewPacketId(suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID, 1)
 	fee := types.NewFee(defaultReceiveFee, defaultAckFee, defaultTimeoutFee)
 
+	var packetFees []types.PacketFee
 	for i := 1; i < 6; i++ {
 		packetFee := types.NewPacketFee(fee, suite.chainA.SenderAccount.GetAddress().String(), nil)
-		suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), packetID, packetFee)
+		packetFees = append(packetFees, packetFee)
```
Comment (seantking) [inline modules/apps/29-fee/keeper/keeper_test.go:104]: Can we do this the same way you have [here](https://github.com/cosmos/ibc-go/pull/1252/files#diff-576b5df34a660fb3ea0475bd98de7e9a9e14617a8d67426801fe84f32b1ce2d8R394)?

i.e.

```
packetFee := types.NewPacketFee(fee, suite.chainA.SenderAccount.GetAddress().String(), nil)
packetFees := []types.PacketFee{packetFee, packetFee, packetFee, packetFee, packetFee}
suite.chainA.GetSimApp().IBCFeeKeeper.SetFeesInEscrow(suite.chainA.GetContext(), packetID, types.NewPacketFees(packetFees))

```

File: modules/apps/29-fee/keeper/escrow_test.go [hunk 2]
```diff
@@ -9,116 +9,6 @@ import (
...
-
-			if tc.expPass {
-				feesInEscrow, found := suite.chainA.GetSimApp().IBCFeeKeeper.GetFeesInEscrow(suite.chainA.GetContext(), packetID)
-				suite.Require().True(found)
-				// check if the escrowed fee is set in state
-				suite.Require().True(feesInEscrow.PacketFees[0].Fee.AckFee.IsEqual(fee.AckFee))
-				suite.Require().True(feesInEscrow.PacketFees[0].Fee.RecvFee.IsEqual(fee.RecvFee))
-				suite.Require().True(feesInEscrow.PacketFees[0].Fee.TimeoutFee.IsEqual(fee.TimeoutFee))
-				// check if the fee is escrowed correctly
-				hasBalance := suite.chainA.GetSimApp().BankKeeper.HasBalance(suite.chainA.GetContext(), suite.chainA.GetSimApp().IBCFeeKeeper.GetFeeModuleAddress(), sdk.Coin{Denom: sdk.DefaultBondDenom, Amount: sdk.NewInt(600)})
```
Comment (seantking) [inline modules/apps/29-fee/keeper/escrow_test.go:108]: I see that we're no longer testing this logic anywhere now. I think we don't want to omit these checks :thinking: 

Kind of a pain in the ass to cover this but I think it's good to have. Maybe if it's not possible to do now we should take it as a sign to leave `EscrowPacketFee` public?

Comment (damiannolan) [reply modules/apps/29-fee/keeper/escrow_test.go:108]: I'll look into making assertions on balances in the msg_server tests 👍

**Review by colin-axner — APPROVED** [review]
ACK pending the additional testing checks in `msg_server_test.go` as requested by @seantking 

Great work! Not a fun pr to take on, but I'm a fan of the code improvement. Makes our tests slightly less nice, but I like having a clear entry point exposed to this functionality. Will prevent external developers from making mistakes. I feel pretty confident that someone would try to directly call `EscrowPacketFee` instead of sending a `PayPacketFee` message - thus bypassing the checks we have in place

File: modules/apps/29-fee/keeper/escrow_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/apps/29-fee/keeper/escrow_test.go:108]: Done! ✅ 

Added assertions on escrow account balances for both success and failure cases!

File: modules/apps/29-fee/keeper/escrow_test.go [hunk 3]
```diff
@@ -229,12 +119,10 @@ func (suite *KeeperTestSuite) TestDistributeFee() {
...
 			packetFee = types.NewPacketFee(fee, refundAcc.String(), []string{})
-			err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), packetID, packetFee)
-			suite.Require().NoError(err)
 
-			// escrow a second packet fee to test with multiple fees distributed
-			err = suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), packetID, packetFee)
-			suite.Require().NoError(err)
+			packetFees := []types.PacketFee{packetFee, packetFee}
+			suite.chainA.GetSimApp().IBCFeeKeeper.SetFeesInEscrow(suite.chainA.GetContext(), packetID, types.NewPacketFees(packetFees))
+			suite.chainA.GetSimApp().BankKeeper.SendCoinsFromAccountToModule(suite.chainA.GetContext(), refundAcc, types.ModuleName, packetFee.Fee.Total().Add(packetFee.Fee.Total()...))
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/29-fee/keeper/escrow_test.go]: Done! Now checking all error returns from this 👍

File: modules/apps/29-fee/keeper/keeper_test.go (hunk 1 above)
Comment (damiannolan) [reply modules/apps/29-fee/keeper/keeper_test.go]: Updated!

File: modules/apps/29-fee/keeper/escrow_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/apps/29-fee/keeper/escrow_test.go:108]: Actually, I just done it in the async handler. I'll push an update for the other one too!

File: modules/apps/29-fee/keeper/escrow_test.go [hunk 4]
```diff
@@ -9,117 +9,7 @@ import (
...
-			originalBal := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
-
-			// escrow the packet fee
-			err = suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), packetID, packetFee)
-
-			if tc.expPass {
-				feesInEscrow, found := suite.chainA.GetSimApp().IBCFeeKeeper.GetFeesInEscrow(suite.chainA.GetContext(), packetID)
-				suite.Require().True(found)
-				// check if the escrowed fee is set in state
-				suite.Require().True(feesInEscrow.PacketFees[0].Fee.AckFee.IsEqual(fee.AckFee))
```
Comment (seantking) [inline modules/apps/29-fee/keeper/escrow_test.go:104]: Should we add these checks somewhere also?

**Review by seantking — APPROVED** [review]
Approved but I noticed we don't check for the stored fees in escrow anymore -> https://github.com/cosmos/ibc-go/pull/1252/files#r850491652

Is this something we want to keep? I get it's somewhat redundant but may be a good safety check. I think it's worth adding.

File: modules/apps/29-fee/keeper/escrow_test.go (hunk 4 above)
Comment (damiannolan) [reply modules/apps/29-fee/keeper/escrow_test.go:104]: Done

**Pull request merged**